File: app/src/main/java/com/joaquinjimenezgarcia/resolucionecuaciones/MainActivity.java
package com.joaquinjimenezgarcia.resolucionecuaciones;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private Button button;
    private EditText valor1, valor2, valor3;
    private TextView textX1, textX2;

    public static double a, b, c, x1, x2;
    public static String valorX1;
    public static String valorX2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        button = (Button) findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                cuenta();
            }
        });
    }

    public void cuenta(){
        valor1 = (EditText) findViewById(R.id.valor1);
        valor2 = (EditText) findViewById(R.id.valor2);
        valor3 = (EditText) findViewById(R.id.valor3);
        textX1 = (TextView) findViewById(R.id.textX1);
        textX2 = (TextView) findViewById(R.id.textX2);

        a = Double.parseDouble(valor1.getText().toString());
        b = Double.parseDouble(valor2.getText().toString());
        c = Double.parseDouble(valor3.getText().toString());

        x1 = (-b + (Math.sqrt(b*b - 4*a*c)))/2*a;
        x2 = (-b - (Math.sqrt(b*b - 4*a*c)))/2*a;

        valorX1 = String.valueOf(x1);
        valorX2 = String.valueOf(x2);

        textX1.setText(valorX1);
        textX2.setText(valorX2);
    }
}

File: app/src/main/java/com/joaquinjimenezgarcia/resolucionecuaciones/MainActivity.kt
package com.joaquinjimenezgarcia.resolucionecuaciones

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    private var button: Button? = null
    private var valor1: EditText? = null
    private var valor2: EditText? = null
    private var valor3: EditText? = null
    private var textX1: TextView? = null
    private var textX2: TextView? = null

    var a: Double = 0.0
    var b: Double = 0.0
    var c: Double = 0.0
    var x1: Double = 0.0
    var x2: Double = 0.0
    var valorX1: String = ""
    var valorX2: String = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button = findViewById(R.id.button) as Button

        button!!.setOnClickListener { cuenta() }
    }

    fun cuenta() {
        valor1 = findViewById(R.id.valor1) as EditText
        valor2 = findViewById(R.id.valor2) as EditText
        valor3 = findViewById(R.id.valor3) as EditText
        textX1 = findViewById(R.id.textX1) as TextView
        textX2 = findViewById(R.id.textX2) as TextView

        a = java.lang.Double.parseDouble(valor1!!.text.toString())
        b = java.lang.Double.parseDouble(valor2!!.text.toString())
        c = java.lang.Double.parseDouble(valor3!!.text.toString())

        x1 = (-b + Math.sqrt(b * b - 4.0 * a * c)) / 2 * a
        x2 = (-b - Math.sqrt(b * b - 4.0 * a * c)) / 2 * a

        valorX1 = x1.toString()
        valorX2 = x2.toString()

        textX1!!.text = valorX1
        textX2!!.text = valorX2
    }
}
